src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(arg) {
+    if (arg === undefined) {
+      const total = sum;
+
+      sum = 0;
+
+      return total;
+    }
+
+    sum += arg;
```

Comment:
The sum of the arguments is being calculated using a simple addition operator. This should be replaced with the `reduce` function for calculating the sum according to the task requirements.